engine/gamesys/src/gamesys/components/comp_gui.cpp

```diff
@@ -306,9 +306,11 @@ namespace dmGameSystem
                     // animation. Instead we can immediately cancel the animation and the node will
                     // still have the correct image.
                     // By doing this we'll not take up an animation slot (there's a max animation cap).
+                    // Even though we cancel animation we still need flipbook hash:
+                    // https://github.com/defold/defold/issues/6551
                     if (dmGui::GetNodeAnimationFrameCount(scene, n) == 1)
                     {
-                        dmGui::CancelNodeFlipbookAnim(scene, n);
+                        dmGui::CancelNodeFlipbookAnim(scene, n, true);
```

Comment:
This logic only uses dmGui functions.

Perhaps we can move this into the `dmGui::PlayNodeFlipbookAnim()`  ?
That way we don't need to deal with it here, and we don't need the weird "keep_anim_hash" argument.

Reply:
Unfortunately we can't.
That was my first assumption I worked on yesterday.
If we do so we break all the one-frame animations because they will never play. In this case we cancel one-frame animation ONLY when we initially setup node.

Reply:
Feels like a good code comment.

Reply:
Done! Thank you!